oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/LastModifiedRange.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.jackrabbit.oak.index.indexer.document;
+
+public class LastModifiedRange {
+
+    private final long lastModifiedLowerBound;
+    private final long lastModifiedUpperBound;
+    private final boolean isUpperBoundExclusive;
+
+    public LastModifiedRange(long lastModifiedLowerBound, long lastModifiedUpperBound) {
+        this(lastModifiedLowerBound, lastModifiedUpperBound, true);
+    }
+
+    public LastModifiedRange(long lastModifiedLowerBound, long lastModifiedUpperBound, boolean isUpperBoundExclusive) {
```

Comment:
I would simplify the logic here and hardcode that isUpperBoundExclusive is always true. You could rename lastModifiedLowerBound => lastModifiedFrom, and lastModifiedUpperBound => lastModifiedTo. This is the logic used in the JDK, e.g. java.util.Arrays.copyOfRange and many other places. Because having the isUpperBoundExclusive flag would mean that we need to add a lot more unit tests for this class - if we implement both behaviours, we need to test both as well.